prov/shm/src/smr_progress.c

```diff
@@ -323,7 +323,7 @@ static int smr_mmap_peer_copy(struct smr_ep *ep, struct smr_cmd *cmd,
 		    != *total_len) {
 			FI_WARN(&smr_prov, FI_LOG_EP_CTRL,
 				"mmap iov copy in error\n");
-			ret = -FI_EIO;
+			ret = -FI_ETRUNC;
```

Comment:
I didn't think it needed to be changed because the mmap protocol is never used with anything other than `FI_HMEM_SYSTEM`:
https://github.com/ofiwg/libfabric/blob/aaa8b8b3ebde73e8246643f8cf8a7597c8373166/prov/shm/src/smr_rma.c#L191-L207
The same logic for protocol selection is used in `smr_generic_sendmsg`

Reply:
This memory type refers to the source type of memory. If memory is going from host->device then the target memory can still be HMEM, which is the one referred to in the progress function.

Reply:
Ah, of course. I will fix that up

Reply:
Is this suggested change made in the latest revision?

Reply:
Pls ignore my above comment, see in your next commit.